### PR TITLE
Pathway vectorstore and rag-pathway template

### PR DESCRIPTION
**Pull request by lewymati**
- **Description:** a description of the change, 
  - **Issue:** the issue # it fixes (if applicable),
  - **Dependencies:** any dependencies required for this change,
  - **Tag maintainer:** for a quicker response, tag the relevant maintainer (see below),
  - **Twitter handle:** we announce bigger features on Twitter. If your PR gets announced, and you'd like a mention, we'll gladly shout you out!

Please make sure your PR is passing linting and testing before submitting. Run `make format`, `make lint` and `make test` to check this locally.

See contribution guidelines for more information on how to write/run tests, lint, etc: 
https://github.com/langchain-ai/langchain/blob/master/.github/CONTRIBUTING.md

If you're adding a new integration, please include:
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/extras` directory.

If no one reviews your PR within a few days, please @-mention one of @baskaryan, @eyurtsev, @hwchase17.

### CODE REVIEW

File: templates/rag-pathway/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2023 LangChain, Inc.
```
Comment (janchorowski) [inline templates/rag-pathway/LICENSE:3]: Can we double-check it?

File: templates/rag-pathway/sample_documents/repo_readme.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+Pathway provides a high-level programming interface in Python for defining data transformations, aggregations, and other operations on data streams.
+With Pathway, you can effortlessly design and deploy sophisticated data workflows that efficiently handle high volumes of data in real time.
+
+Pathway is interoperable with various data sources and sinks such as Kafka, CSV files, SQL/noSQL databases, and REST API's, allowing you to connect and process data from different storage systems.
+
+Typical use-cases of Pathway include realtime data processing, ETL (Extract, Transform, Load) pipelines, data analytics, monitoring, anomaly detection, and recommendation. Pathway can also independently provide the backbone of a light LLMops stack for real-time LLM applications.
```
Comment (dxtrous) [inline templates/rag-pathway/sample_documents/repo_readme.md:12]:
```suggestion
Pathway is an open framework for high-throughput and low-latency real-time data processing. It is used to create Python code which seamlessly combines batch processing, streaming, and real-time API's for LLM apps. Pathway's distributed runtime (🦀-🐍) provides fresh results of your data pipelines whenever new inputs and requests are received.

Pathway provides a high-level programming interface in Python for defining data transformations, aggregations, and other operations on data streams.
Pathway is interoperable with various data sources and sinks such as Kafka, CSV files, SQL/noSQL databases, and REST API's, allowing you to connect and process data from different storage systems.
```

File: templates/rag-pathway/sample_documents/repo_readme.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
+Pathway is an open framework for high-throughput and low-latency real-time data processing. It is used to create Python code which seamlessly combines batch processing, streaming, and real-time API's for LLM apps. Pathway's distributed runtime (🦀-🐍) provides fresh results of your data pipelines whenever new inputs and requests are received.
+
+Pathway is an incremental data stream processing engine
+
```
Comment (dxtrous) [inline templates/rag-pathway/sample_documents/repo_readme.md]:
```suggestion
```

File: docs/docs/integrations/vectorstores/pathway.ipynb [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    " ## Running in a separate process\n",
+    " A production deployment will typically run the server in a separate process. We provide a template application under [`templates`](https://github.com/langchain-ai/langchain/tree/master/templates/rag-pathway). We recommend to run the Pathway data indexing pipeline in a container-based deployment environment, such as Docker or Kubernetes. For more info see [Pathway's deployment guide](https://pathway.com/developers/user-guide/deployment/).\n",
+    "\n",
+    " ## Configuring the cache\n",
+    " The Pathway vectorizing pipeline comes with an embeddings cache:\n",
+    " ```python\n",
+    " vector_server.run_server(..., with_cache=True)\n",
+    " ```\n",
+    "\n",
+    " The default cache configuration is the locally hosted disk cache, stored in the `./Cache` directory. However, it is possible to customize it by explicitly specifying the caching backend that can be chosen among several persistent backends [options](/developers/api-docs/persistence-api/#pathway.persistence.Backend)."
```
Comment (janchorowski) [inline docs/docs/integrations/vectorstores/pathway.ipynb]:
```suggestion
    " The default cache configuration is the locally hosted disk cache, stored in the `./Cache` directory. However, it is possible to customize it by explicitly specifying the caching backend that can be chosen among several persistent backends [options](https://pathway.com/developers/api-docs/persistence-api/#pathway.persistence.Backend)."
```

**Pull request merged**